src/Angular.js

```diff
@@ -334,11 +334,11 @@ function baseExtend(dst, objs, deep) {
       var key = keys[j];
       var src = obj[key];
 
-      if (deep && isObject(src)) {
+      if (deep && isObject(src) && !isDate(src)) {
         if (!isObject(dst[key])) dst[key] = isArray(src) ? [] : {};
         baseExtend(dst[key], [src], true);
       } else {
-        dst[key] = src;
+        dst[key] = isDate(src) ? new Date(src.valueOf()) : src;
```

Comment:
I don't think this is the right think to do. It will also apply to non-deep copying, where I believe it would be better to just assign the same Date object.

It's OK fr deep extending I guess, so maybe something like:

```js
       if (deep && isObject(src)) {
         if (isDate(src)) {
           dst[key] = new Date(src.valueOf());
         } else {
           if (!isObject(dst[key])) dst[key] = isArray(src) ? [] : {};
           baseExtend(dst[key], [src], true);
         }
       } else {
         dst[key] = src;
       }
```